### DISCUSSION TITLE
Understanding scenicPlus settings and Outputs

### DISCUSSION THREAD

**Discussion by orian116** [aertslab/scenicplus#478] [Q&A]
Hello. Thanks so much for the package! I had some basic questions about the package:

1. Could you explain how scenicplus creates the pseudomultiome dataset? I recognize it's not a 1:1 relationship between the **key_to_group_by** set in the config.yaml file and the resulting AUC matrix has much less cells than either ATAC or RNA-seq matrices. Also on that, how did you decide the default **nr_cells_per_metacells**  to be 10 and how does increasing or decreasing that number affect the results?
2. I had some general questions about what the "Regions" in the eRegulon represent and how they are related to the TF and gene in terms of location on the genome. Does Scenicplus weigh differently TFs that regulate genes/regions on different chromosomes i.e TFS that are trans acting elements? Are these trans acting elements lower confidence than cis acting TFs i.e  TFs that regulate the genes/regions on the same chromosome?

**Comment by SeppeDeWinter** [comment]
Hi @orian116 

> Could you explain how scenicplus creates the pseudomultiome dataset? 

I randomly samples cells from the scRNA-seq and scATAC-seq side and averages their gene expression/chromatin accessibility profile to create a "metacell", by default this uses 10 cells per metacell.

> Also on that, how did you decide the default nr_cells_per_metacells to be 10 and how does increasing or decreasing that number affect the results?

The default of 10 is quite arbitrary, we did not test this extensively.

> I had some general questions about what the "Regions" in the eRegulon represent and how they are related to the TF and gene in terms of location on the genome. Does Scenicplus weigh differently TFs that regulate genes/regions on different chromosomes i.e TFS that are trans acting elements? Are these trans acting elements lower confidence than cis acting TFs i.e TFs that regulate the genes/regions on the same chromosome?

I'm not sure wether I'm understanding this question correctly. Could you rephrase? 

All the best,

Seppe

**Reply by PhrenoVermouth** [reply]
> Hi @orian116
> 
> > Could you explain how scenicplus creates the pseudomultiome dataset?
> 
> I randomly samples cells from the scRNA-seq and scATAC-seq side and averages their gene expression/chromatin accessibility profile to create a "metacell", by default this uses 10 cells per metacell.
> 
> > Also on that, how did you decide the default nr_cells_per_metacells to be 10 and how does increasing or decreasing that number affect the results?
> 
> The default of 10 is quite arbitrary, we did not test this extensively.
> 
> > I had some general questions about what the "Regions" in the eRegulon represent and how they are related to the TF and gene in terms of location on the genome. Does Scenicplus weigh differently TFs that regulate genes/regions on different chromosomes i.e TFS that are trans acting elements? Are these trans acting elements lower confidence than cis acting TFs i.e TFs that regulate the genes/regions on the same chromosome?
> 
> I'm not sure wether I'm understanding this question correctly. Could you rephrase?
> 
> All the best,
> 
> Seppe

Hey @SeppeDeWinter ! Thanks for your package! After reading for so many issues and discussions, including the tutorials, I'm still so confused by the para 'key_to_group_by'. I understand this is a para designed for pseudo-multiome. From my intuition, if you want to align two modalities by cell annotation, we should declare related meta cols in both GEX and ATAC, right? But why only one of them is required as below? ` This variable should be prefixed with eiter “GEX:” or “ACC:”.`

**Reply by PhrenoVermouth** [reply]
And besides, from [the previous non-multiome example ](https://github.com/aertslab/scenicplus/blob/old/notebooks/mix_melanoma_cell_lines.ipynb) it seems the prefix is not required? Maybe the rule has changed as the development goes on? Thanks for clarifying!

**Reply by SeppeDeWinter** [reply]
Hi @PhrenoVermouth 

The `key_to_group_by` represents a cell-metadata column in both your gene expression and chromatin accessibility data which is used to group both gene expression and chromatin accessibility cells by to generate pseudo-multiome data.

For example this can be the cell type annotation. When cells in both the gene expression and chromatin accessibility data are annotated the same, you can use this annotation to combine similar cells across both modalities.

The prefix you are talking about is automatically added to your data when you run it through the pipeline.

I hope this answers your question?

All the best,

S

**Reply by PhrenoVermouth** [reply]
Yeah I'd ensured the input from both modalities have the same annotation obs name, and it worked fine. I did not manually add a prefix to it. Thanks for your reply!
